### ISSUE TITLE
xdpyinfo check failed inside the mahimahi container

### ISSUE THREAD

**Issue by BobAnkh** [ravinet/mahimahi#138]
I have a problem when using mahimahi on Ubuntu 18.04.4LTS(GNU/Linux 4.15.0-64-generic x86_64) with Python=3.7.5. I want to use pyvirtualdisplay inside the mahimahi container but failed.

First I run `mm-delay 40` . Then I run `python test.py` .
The code in the test.py is as follows:
```python
  from pyvirtualdisplay import Display
  from time import sleep
  display = Display(visible = 1,size = (800,600))
  display.start()
  sleep(60)
  display.stop()
  print('done')
```
I failed with:

```
Traceback (most recent call last):
  File "test.py", line 4, in <module>
    display.start()
  File "/usr/local/lib/python3.7/dist-packages/pyvirtualdisplay/abstractdisplay.py", line 171, in start
    raise XStartTimeoutError(msg % d)
pyvirtualdisplay.abstractdisplay.XStartTimeoutError: Failed to start X on display ":2751" (xdpyinfo check failed).
```

The code of pyvirtualdisplay.abstractdisplay shows that it runs `exit_code=EasyProcess('xdpyinfo').call().return_code`, the value of `exit_code` is 1 inside the mahimahi container while the value of 'exit_code' is 0 outside.
Also I directly run `xdpyinfo` in the mahimahi container but only returned with this:
```
 xdpyinfo:  unable to open display "localhost:10.0".
```

So can anybody help me to solve this problem? Thank you very much!

**Comment by keithw** [comment]
I think there are a few (different) things going on here...

(1) From the "localhost:10.0" (the contents of your DISPLAY environment variable), it looks like you are using OpenSSH X forwarding to log in to this machine, instead of sitting directly in front of it. OpenSSH listens on a TCP socket (localhost port 6010, corresponding to display ":10") for incoming connections from X clients (like xdpyinfo, or xterm, or firefox, etc.). But it's listening on localhost:6010 **outside** the mahimahi container. So the X clients that run inside can't reach your SSH server, which means they can't be forwarded over your SSH connection. You wouldn't have that particular problem if you were sitting in front of the machine itself.

(2) None of that should really matter because pyvirtualdisplay is supposed to be starting its own "virtual" X server and then testing that it got started by running xdpyinfo with some DISPLAY environment variable that points to the new virtual X server (not your "real" X server). They have several different "virtual" X servers that they can start (Xvfb, Xephyr, Xvnc), and I'm not sure which one you're trying to start here.

I just tried with Xvfb manually inside a mm-delay container and it seemed to work fine (I set my DISPLAY environment variable to the screen number of the Xvfb and then ran xdpyinfo and it all seemed to work fine).

So my guess is that the "localhost:10.0" is a red herring. (You might try unsetting your DISPLAY environment variable completely -- if pyvirtualdisplay is well-implemented, it shouldn't care and should keep working in the settings it was working in before.) Why pyvirtualdisplay is having trouble inside mm-display probably depends on what kind of virtual display you're using it to start. You might better off asking the pyvirtualdisplay folks about this and seeing if they can help.
